packages/blocks/microsoft-outlook/src/lib/triggers/new-email.ts

```diff
@@ -0,0 +1,218 @@
+import { Client, PageCollection } from '@microsoft/microsoft-graph-client';
+import { Message, Recipient } from '@microsoft/microsoft-graph-types';
+import { DedupeStrategy, Polling, pollingHelper } from '@openops/blocks-common';
+import {
+  BlockPropValueSchema,
+  Property,
+  TriggerStrategy,
+  createTrigger,
+} from '@openops/blocks-framework';
+import dayjs from 'dayjs';
+import { microsoftOutlookAuth } from '../common/auth';
+import { mailFolderIdDropdown } from '../common/props';
+
+function normalizeString(value: unknown): string {
+  return String(value).toLowerCase().trim();
+}
+
+function isValidFilterArray(value: unknown): value is unknown[] {
+  return Array.isArray(value) && value.length > 0;
+}
+
+function normalizeFilterArray(array: unknown[]): string[] {
+  return array.filter(Boolean).map(normalizeString);
+}
+
+function extractEmailAddresses(
+  recipients: Recipient[] | null | undefined,
+): string[] {
+  if (!recipients) return [];
+  return recipients.map((recipient) =>
+    normalizeString(recipient.emailAddress?.address || ''),
+  );
+}
+
+function matchesAnyFilter(targets: string[], filters: string[]): boolean {
+  return filters.some((filter) =>
+    targets.some((target) => target.includes(filter)),
+  );
+}
+
+function applyClientSideFilters(
```

Comment:
i really wanted to do server side filtering but i just couldnt get it to work, mainly theres and issue because using the lastDate logic with filtering by email, just doesnt work easily, one uss $search while the other uses $filter, and you cannot use those two together in microsoft graph